### DISCUSSION TITLE
Github Actions to make changes to a file

### DISCUSSION THREAD

**Discussion by allisonsampaio** [community/community#26842] [Actions]
<p>Hello, I am learning to use GitHub Actions for a project and I am having difficulties.</p>
<p>How can I use Github Actions to make changes to a file that is in my repository?<br>
I’m doing a gamification project that assigns points to users who commit or other actions in my repository, these points are in a JSON file linked to the user’s name, when the user performs a certain action (such as a commit) this value must be added to that file.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/allisonsampaio">@allisonsampaio</a> ,</p>
<p>To chang the files in the repository, generally you need to setup the steps like as below:</p>
<ol>
<li>
<p>Use the <a href="https://github.com/actions/checkout" rel="noopener nofollow ugc"><strong>checkout</strong></a> action to check out the repository to the runner machine.</p>
</li>
<li>
<p>Execute the shell scripts or actions to modify the files.</p>
<ul>
<li>The shell scripts just is like as what you can run in the local machines.</li>
<li>Abuout the actions, you can go to the <a href="https://github.com/marketplace?type=actions" rel="noopener nofollow ugc"><strong>GitHub Marketplace</strong></a> to look for the related actions that can modify the specific types of files that you want to change. If you do not find any available action, you can reference to “<a href="https://docs.github.com/en/actions/creating-actions" rel="noopener nofollow ugc"><strong>Creating actions</strong></a>” to create yourself customized action.</li>
</ul>
</li>
<li>
<p>Run the ralated git commands to commit and push the changes to the repository on GitHub.</p>
<ul>
<li><a href="https://git-scm.com/docs/git-config" rel="noopener nofollow ugc"><strong>git config</strong></a></li>
<li><a href="https://git-scm.com/docs/git-add" rel="noopener nofollow ugc"><strong>git add</strong></a></li>
<li><a href="https://git-scm.com/docs/git-commit" rel="noopener nofollow ugc"><strong>git commit</strong></a></li>
<li><a href="https://git-scm.com/docs/git-push" rel="noopener nofollow ugc"><strong>git push</strong></a></li>
</ul>
</li>
</ol>
<p>A simple demo:</p>
<pre><code class="lang-yaml">jobs:
  job1:
    name: Modifiy repository files
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v2.3.2

      - name: Modifiy the specific files using scripts
        run: |
          --- shell scripts or command lines to execute ---
# If you execute some shell scripts or command lines to modify
# the files, use the 'run' key.
# More details, see "https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun"

      - name: Modifiy the specific files using action
        uses: action-owner/action-repo@ref
# If you execute an action to modify the files, use the 'uses' key.
# Just like as the previous step that executes the checkout action.
# More details, see "https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses"

      - name: Commit and push changes
        run: |
          git config --global user.name "your username"
          git config --global user.email "your email"

          git add -A
          git commit -m "commit message"
          git push
</code></pre>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/allisonsampaio">@allisonsampaio</a> ,<br>
How are things going?<br>
Is my above suggestion helpful to you?<br>
Please try it, and feel free to tell me if you have any update.</p>

**Comment by allisonsampaio** [comment]
<p>Hi <a class="mention" href="/brightran">@brightran</a><br>
It helped me a lot, thank you</p>
<p>Do you know how I can access the event data? (as the text of an issue_comment event)</p>

**Comment by Simran-B** [comment]
<p>You can add a step like this for debugging purposes (assuming bash as shell):</p>
<pre><code class="lang-auto">- name: GitHub Event
  run: |
    cat &lt;&lt; 'EOF'
      ${{ toJSON(github.event) }}
    EOF
</code></pre>
<p>If you want to react to issue comments, you could trigger a workflow like:</p>
<pre><code class="lang-auto">on:
  issue_comment:
</code></pre>
<p>Note that this won’t be run for the initial post but only subsequent comments.</p>
<p>To access the raw text of the comment (in Markdown syntax, not converted to HTML), you can use:</p>
<pre><code class="lang-auto">${{ github.event.comment.body }}
</code></pre>
<p>If the event sub-type is <code>edited</code>, then you can also access the previous text with <code>github.event.changes.body.from</code>.</p>
<details>
<summary>
Full event object example</summary>
<pre><code class="lang-auto">  {
  "action": "edited",
  "changes": {
    "body": {
      "from": "### Test\r\n\r\nHello!\r\n\r\n&gt; quote\r\n&gt; goes here\r\n\r\n**Bold** _italic_ ~strike~ `code`\r\n\r\n- foo\r\n- bar\r\n\r\n1. One\r\n2. Two\r\n\r\n---\r\n\r\n```js\r\nconsole.log(\"code block\");\r\n```\r\n\r\nMention: @&lt;UserName&gt; \r\n\r\n[Link](http://example.org)"
    }
  },
  "comment": {
    "author_association": "OWNER",
    "body": "### Test\r\n\r\nHello!\r\n\r\n&gt; quote\r\n&gt; goes here\r\n\r\n**Bold** _italic_ ~strike~ `code`\r\n\r\n- foo\r\n- bar\r\n\r\n1. One\r\n2. Two\r\n\r\n---\r\n\r\n```js\r\nconsole.log(\"code block\");\r\n```\r\n\r\nMention: @&lt;UserName&gt; \r\n\r\n[Link](http://example.org)\r\n",
    "created_at": "2020-09-27T19:59:45Z",
    "html_url": "https://github.com/&lt;UserName&gt;/&lt;Repo&gt;/issues/4#issuecomment-987654321",
    "id": 987654321,
    "issue_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/4",
    "node_id": "xYzABcdEc3VlQ29tbWVudDY5OTY4MTE1Ng==",
    "performed_via_github_app": null,
    "updated_at": "2020-09-27T20:57:13Z",
    "url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/comments/987654321",
    "user": {
      "avatar_url": "https://avatars3.githubusercontent.com/u/1234567?v=4",
      "events_url": "https://api.github.com/users/&lt;UserName&gt;/events{/privacy}",
      "followers_url": "https://api.github.com/users/&lt;UserName&gt;/followers",
      "following_url": "https://api.github.com/users/&lt;UserName&gt;/following{/other_user}",
      "gists_url": "https://api.github.com/users/&lt;UserName&gt;/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/&lt;UserName&gt;",
      "id": 1234567,
      "login": "&lt;UserName&gt;",
      "node_id": "xYzABcdEcjc4MTk5OTE=",
      "organizations_url": "https://api.github.com/users/&lt;UserName&gt;/orgs",
      "received_events_url": "https://api.github.com/users/&lt;UserName&gt;/received_events",
      "repos_url": "https://api.github.com/users/&lt;UserName&gt;/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/&lt;UserName&gt;/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/&lt;UserName&gt;/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/&lt;UserName&gt;"
    }
  },
  "issue": {
    "active_lock_reason": null,
    "assignee": {
      "avatar_url": "https://avatars3.githubusercontent.com/u/1234567?v=4",
      "events_url": "https://api.github.com/users/&lt;UserName&gt;/events{/privacy}",
      "followers_url": "https://api.github.com/users/&lt;UserName&gt;/followers",
      "following_url": "https://api.github.com/users/&lt;UserName&gt;/following{/other_user}",
      "gists_url": "https://api.github.com/users/&lt;UserName&gt;/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/&lt;UserName&gt;",
      "id": 1234567,
      "login": "&lt;UserName&gt;",
      "node_id": "xYzABcdEcjc4MTk5OTE=",
      "organizations_url": "https://api.github.com/users/&lt;UserName&gt;/orgs",
      "received_events_url": "https://api.github.com/users/&lt;UserName&gt;/received_events",
      "repos_url": "https://api.github.com/users/&lt;UserName&gt;/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/&lt;UserName&gt;/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/&lt;UserName&gt;/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/&lt;UserName&gt;"
    },
    "assignees": [
      {
        "avatar_url": "https://avatars3.githubusercontent.com/u/1234567?v=4",
        "events_url": "https://api.github.com/users/&lt;UserName&gt;/events{/privacy}",
        "followers_url": "https://api.github.com/users/&lt;UserName&gt;/followers",
        "following_url": "https://api.github.com/users/&lt;UserName&gt;/following{/other_user}",
        "gists_url": "https://api.github.com/users/&lt;UserName&gt;/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/&lt;UserName&gt;",
        "id": 1234567,
        "login": "&lt;UserName&gt;",
        "node_id": "xYzABcdEcjc4MTk5OTE=",
        "organizations_url": "https://api.github.com/users/&lt;UserName&gt;/orgs",
        "received_events_url": "https://api.github.com/users/&lt;UserName&gt;/received_events",
        "repos_url": "https://api.github.com/users/&lt;UserName&gt;/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/&lt;UserName&gt;/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/&lt;UserName&gt;/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/&lt;UserName&gt;"
      }
    ],
    "author_association": "OWNER",
    "body": "`issue_comment` event really just triggers for comments, no the initial post.",
    "closed_at": null,
    "comments": 1,
    "comments_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/4/comments",
    "created_at": "2020-09-27T18:10:01Z",
    "events_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/4/events",
    "html_url": "https://github.com/&lt;UserName&gt;/&lt;Repo&gt;/issues/4",
    "id": 709787574,
    "labels": [
      {
        "color": "e4e669",
        "default": true,
        "description": "This doesn't seem right",
        "id": 2067629346,
        "name": "invalid",
        "node_id": "xYzABcdEZWwyMDY3NjI5MzQ2",
        "url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/labels/invalid"
      }
    ],
    "labels_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/4/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "xYzABcdEdWU3MDk3ODc1NzQ=",
    "number": 4,
    "performed_via_github_app": null,
    "repository_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;",
    "state": "open",
    "title": "Issue comment action test",
    "updated_at": "2020-09-27T20:57:13Z",
    "url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/4",
    "user": {
      "avatar_url": "https://avatars3.githubusercontent.com/u/1234567?v=4",
      "events_url": "https://api.github.com/users/&lt;UserName&gt;/events{/privacy}",
      "followers_url": "https://api.github.com/users/&lt;UserName&gt;/followers",
      "following_url": "https://api.github.com/users/&lt;UserName&gt;/following{/other_user}",
      "gists_url": "https://api.github.com/users/&lt;UserName&gt;/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/&lt;UserName&gt;",
      "id": 1234567,
      "login": "&lt;UserName&gt;",
      "node_id": "xYzABcdEcjc4MTk5OTE=",
      "organizations_url": "https://api.github.com/users/&lt;UserName&gt;/orgs",
      "received_events_url": "https://api.github.com/users/&lt;UserName&gt;/received_events",
      "repos_url": "https://api.github.com/users/&lt;UserName&gt;/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/&lt;UserName&gt;/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/&lt;UserName&gt;/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/&lt;UserName&gt;"
    }
  },
  "repository": {
    "archive_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/{archive_format}{/ref}",
    "archived": false,
    "assignees_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/assignees{/user}",
    "blobs_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/git/blobs{/sha}",
    "branches_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/branches{/branch}",
    "clone_url": "https://github.com/&lt;UserName&gt;/&lt;Repo&gt;.git",
    "collaborators_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/collaborators{/collaborator}",
    "comments_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/comments{/number}",
    "commits_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/commits{/sha}",
    "compare_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/compare/{base}...{head}",
    "contents_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/contents/{+path}",
    "contributors_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/contributors",
    "created_at": "2020-05-17T15:08:58Z",
    "default_branch": "master",
    "deployments_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/deployments",
    "description": null,
    "disabled": false,
    "downloads_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/downloads",
    "events_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/events",
    "fork": false,
    "forks": 0,
    "forks_count": 0,
    "forks_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/forks",
    "full_name": "&lt;UserName&gt;/&lt;Repo&gt;",
    "git_commits_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/git/commits{/sha}",
    "git_refs_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/git/refs{/sha}",
    "git_tags_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/git/tags{/sha}",
    "git_url": "git://github.com/&lt;UserName&gt;/&lt;Repo&gt;.git",
    "has_downloads": true,
    "has_issues": true,
    "has_pages": false,
    "has_projects": true,
    "has_wiki": true,
    "homepage": null,
    "hooks_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/hooks",
    "html_url": "https://github.com/&lt;UserName&gt;/&lt;Repo&gt;",
    "id": 264693846,
    "issue_comment_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/comments{/number}",
    "issue_events_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues/events{/number}",
    "issues_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/issues{/number}",
    "keys_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/keys{/key_id}",
    "labels_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/labels{/name}",
    "language": "Shell",
    "languages_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/languages",
    "license": null,
    "merges_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/merges",
    "milestones_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/milestones{/number}",
    "mirror_url": null,
    "name": "&lt;Repo&gt;",
    "node_id": "xYzABcdEcG9zaXRvcnkyNjQ2OTM4NDY=",
    "notifications_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/notifications{?since,all,participating}",
    "open_issues": 1,
    "open_issues_count": 1,
    "owner": {
      "avatar_url": "https://avatars3.githubusercontent.com/u/1234567?v=4",
      "events_url": "https://api.github.com/users/&lt;UserName&gt;/events{/privacy}",
      "followers_url": "https://api.github.com/users/&lt;UserName&gt;/followers",
      "following_url": "https://api.github.com/users/&lt;UserName&gt;/following{/other_user}",
      "gists_url": "https://api.github.com/users/&lt;UserName&gt;/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/&lt;UserName&gt;",
      "id": 1234567,
      "login": "&lt;UserName&gt;",
      "node_id": "xYzABcdEcjc4MTk5OTE=",
      "organizations_url": "https://api.github.com/users/&lt;UserName&gt;/orgs",
      "received_events_url": "https://api.github.com/users/&lt;UserName&gt;/received_events",
      "repos_url": "https://api.github.com/users/&lt;UserName&gt;/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/&lt;UserName&gt;/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/&lt;UserName&gt;/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/&lt;UserName&gt;"
    },
    "private": false,
    "pulls_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/pulls{/number}",
    "pushed_at": "2020-09-27T20:57:01Z",
    "releases_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/releases{/id}",
    "size": 88,
    "ssh_url": "git@github.com:&lt;UserName&gt;/&lt;Repo&gt;.git",
    "stargazers_count": 0,
    "stargazers_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/stargazers",
    "statuses_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/statuses/{sha}",
    "subscribers_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/subscribers",
    "subscription_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/subscription",
    "svn_url": "https://github.com/&lt;UserName&gt;/&lt;Repo&gt;",
    "tags_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/tags",
    "teams_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/teams",
    "trees_url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;/git/trees{/sha}",
    "updated_at": "2020-09-27T20:57:03Z",
    "url": "https://api.github.com/repos/&lt;UserName&gt;/&lt;Repo&gt;",
    "watchers": 0,
    "watchers_count": 0
  },
  "sender": {
    "avatar_url": "https://avatars3.githubusercontent.com/u/1234567?v=4",
    "events_url": "https://api.github.com/users/&lt;UserName&gt;/events{/privacy}",
    "followers_url": "https://api.github.com/users/&lt;UserName&gt;/followers",
    "following_url": "https://api.github.com/users/&lt;UserName&gt;/following{/other_user}",
    "gists_url": "https://api.github.com/users/&lt;UserName&gt;/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/&lt;UserName&gt;",
    "id": 1234567,
    "login": "&lt;UserName&gt;",
    "node_id": "xYzABcdEcjc4MTk5OTE=",
    "organizations_url": "https://api.github.com/users/&lt;UserName&gt;/orgs",
    "received_events_url": "https://api.github.com/users/&lt;UserName&gt;/received_events",
    "repos_url": "https://api.github.com/users/&lt;UserName&gt;/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/&lt;UserName&gt;/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/&lt;UserName&gt;/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/&lt;UserName&gt;"
  }
}
</code></pre>
</details>
